services/shhext/service.go

```diff
@@ -296,6 +298,25 @@ func (t *tracker) handleEventMailServerRequestCompleted(event whisper.EnvelopeEv
 	}
 }
 
+func (t *tracker) handleEventMailServerRequestFailed(event whisper.EnvelopeEvent) {
+	t.mu.Lock()
+	defer t.mu.Unlock()
+
+	state, ok := t.cache[event.Hash]
+	if !ok || state != MailServerRequestSent {
+		return
```

Comment:
@mandrigin  in case the request has already expired, `ok` would be `false`. shall we send the `SendMailServerRequestFailed` signal anyway?

Reply:
hmm, good point, I think we should